### PR TITLE
Added Rest Api Swift Extension (WIP)

### PR DESCRIPTION
**Pull request by trooper2013**
This PR is the first of the few PR's to follow for adding Swift Extensions to the mobile SDK. Refines some of the Blocks API(s) and introduces an alternative API with a single completionBlock and ResultType.

Todo: Add some RestClient Swift tests.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3261,13 +3275,15 @@
...
 				SWIFT_COMPILATION_MODE = wholemodule;
+				SWIFT_INSTALL_OBJC_HEADER = NO;
 			};
 			name = Release;
 		};
 		82D4645219C7C8180006BDFE /* Debug */ = {
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = CEBA94EF1C1B5D4C00C0B2E0 /* SalesforceSDKCore-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
+				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Is this required? I don't think the framework should embed `Swift`.

Comment (trooper2013) [reply libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: @bhariharan Its not required. Im not sure why Xcode added this. I will look into it and remove it.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    
+    /// Execute a searchResultLayout
+    /// - Parameter objectList: Array of objects.
+    /// - Parameter completionBlock: The completion block to invoke.
+    func searchResultLayout(_ objectList:[String],_ completionBlock: @escaping (Result<RestResponse,RestClientError>) -> () ) -> RestRequest {
+        let request = self.request(forSearchResultLayout: objectList.joined(separator: ","), apiVersion: self.apiVersion)
+        self.send(request: request, completionBlock)
+        return request
+    }
+}
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift]: What about batch / composite and sobject tree?

Comment (trooper2013) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift]: good question. I think I completely missed that.

**Comment by trooper2013** [comment]
Will redo. Sorry Guys. My changes should be been easier to replay from dev

**Pull request closed without merging**